index.bs

```diff
@@ -4572,6 +4572,55 @@ and similarly, closing or aborting the <code>writable</code> side will implicitl
 
 </div>
 
+<h3 id="example-ts-promise-to-readable">Using an identity transform stream as a primitive when manipulating streams</h3>
+
+Combining an identity transform stream with {{pipeTo()}} is a powerful way to manipulate streams. For example,
+concatenating multiple readable streams into one:
+
+<pre><code class="lang-javascript">
+function concatenateReadables(readables) {
+  const ts = new TransformStream();
+  let promise = Promise.resolve();
+  for (const readable of readables) {
+    promise = promise
+        .then(() => readable.pipeTo(ts.writable, { preventClose: true }),
+              reason => {
+                return Promise.all([
+                  ts.writable.abort(reason),
+                  readable.cancel(reason)]);
+              });
+  }
+  promise.then(() => ts.writable.getWriter().close(),
+               reason => ts.writable.abort(reason))
+      .catch(() => {});
+  return ts.readable;
+}
+</code></pre>
+
+The error handling here is subtle because cancelling the concatenated stream has to cancel all the intput
+streams. However, the success case is simple enough. We just pipe each stream in the <code>readables</code> array one at
```

Comment:
I think it might be better to not cancel them, but leave them for someone else to consume. Which may eliminate the catch block, or at least part of it.

Especially in an example.

Reply:
It's not really helpful to leave some of them open, as the caller has no way to detect which ones have been closed.

Since this is an important primitive, I'd like to have a fully correct implementation, or no implementation at all. I expect in the future we will have a static method called `ReadableStream.concatenate()` or similar that does this exact thing.